ibis/backends/postgres/__init__.py

```diff
@@ -144,69 +294,102 @@ def list_tables(self, like=None, schema=None):
             types of `table`.
             :::
         """
-        tables = self.inspector.get_table_names(schema=schema)
-        views = self.inspector.get_view_names(schema=schema)
-        return self._filter_with_like(tables + views, like)
+        if database is not None:
+            util.warn_deprecated(
+                "database",
+                instead="Use the `schema` keyword argument instead",
+                as_of="7.1",
+                removed_in="8.0",
+            )
+
+        conditions = [TRUE]
+
+        if database is not None:
+            conditions = C.table_catalog.eq(sge.convert(database))
+
+        if schema is not None:
+            conditions = C.table_schema.eq(sge.convert(schema))
+
+        col = "table_name"
+        sql = (
+            sg.select(col)
+            .from_(sg.table("tables", db="information_schema"))
+            .distinct()
+            .where(*conditions)
+            .sql(self.name, pretty=True)
```

Comment:
Does this need to be pretty?

Reply:
Probably not, but was easier for debugging

Reply:
Removed.